packages/flutter/lib/src/rendering/mouse_cursor.dart

```diff
@@ -0,0 +1,270 @@
+/// [InkWell.mouseCursor].
+///
+/// {@tool snippet --template=stateless_widget_material}
+/// This sample creates a rectangular region that is wrapped by a [MouseRegion]
+/// with a system mouse cursor. The mouse pointer becomes an I-beam when
+/// hovering the region.
+///
+/// ```dart imports
+/// import 'package:flutter/widgets.dart';
+/// import 'package:flutter/gestures.dart';
+/// ```
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Center(
+///     child: MouseRegion(
+///       cursor: SystemMouseCursors.text,
+///       child: Container(
+///         width: 200,
+///         height: 100,
+///         decoration: BoxDecoration(
+///           color: Colors.blue,
+///           border: Border.all(color: Colors.yellow),
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// Assigning regions with mouse cursors on platforms that do not support mouse
+/// cursors, or when there are no mouse connected, will result in no-op.
+///
+/// ## Related classes
+///
+/// The following classes are designed to be created and used by the framework,
+/// therefore should not be directly used by widgets.
+///
+/// [MouseTrackerCursorMixin] is a class that manages states, and dispatches
+/// specific operations based on general mouse device updates.
+///
+/// [MouseCursorController] implements low-level imperative control by directly
+/// talking to the platform.
+///
+/// See also:
+///
+///  * [MouseRegion], which is a common way of assigning a region with a
+///    [MouseCursor].
+///  * [MouseTracker], which determines the cursor that each device should show,
+///    and dispatches the changing callbacks.
+///  * [SystemMouseCursors], which enumerates supported system cursors.
+@immutable
+abstract class MouseCursor with Diagnosticable {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const MouseCursor();
+
+  /// A very short pretty description of the mouse cursor.
+  ///
+  /// The [debugDescription] shoule be a few words that can differentiate
+  /// instances of a class to make debug information more readable. For example,
+  /// a [SystemMouseCursor] class with description "drag" will be printed as
+  /// "SystemMouseCursor(drag)".
+  ///
+  /// The [debugDescription] should not be null, but can be an empty string,
+  /// which means the class is not configurable.
+  String get debugDescription;
+
+  @override
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.info}) {
+    final String debugDescription = this.debugDescription;
+    if (minLevel.index >= DiagnosticLevel.info.index && debugDescription != null)
+      return '$runtimeType($debugDescription)';
+    return super.toString(minLevel: minLevel);
+  }
+}
+
+/// An interface for mouse cursors that have all resources prepared and ready to
+/// be sent to the system.
```

Comment:
```suggestion
/// be sent to the engine for rendering.
```
(Or whatever seems more clear). "System" just seems kind of generic.  Maybe Host OS? Platform?

Reply:
What about
```dart
/// An interface for mouse cursors that have all resources prepared and ready to
/// be used by the operating system.
```